frontend/lib/shared/payment_details_page.dart

```diff
@@ -0,0 +1,181 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:flutter_starter/l10n/app_localizations.dart';
+import 'package:flutter_starter/shared/grid.dart';
+import 'package:flutter_starter/shared/json_schema_form.dart';
+import 'package:flutter_starter/shared/payment_method.dart';
+import 'package:flutter_starter/shared/search_payment_methods_page.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+class PaymentDetailsPage extends HookConsumerWidget {
+  const PaymentDetailsPage({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final paymentMethods = ref.watch(paymentMethodProvider);
+
+    final paymentTypes = paymentMethods
+        ?.map((method) => method.kind.split('_').firstOrNull)
+        .toSet();
+
+    final selectedPaymentMethod = useState(paymentMethods?.firstOrNull);
```

Comment:
```
final selectedPaymentMethod = useState<PaymentMethod?>(null);
```
assigning `selectedPaymentMethod` to null would make more sense here since there is a `useEffect()`?

Reply:
I don't think it hurts anything, but this logic is a bit tricky to follow. Is there a way to refactor this to make it more obvious what's happening? We can also refactor later if needed.